### DISCUSSION TITLE
Image conversion AdobeRGB .jpg to sRGB .png is Grayscale

### DISCUSSION THREAD

**Discussion by pixelzooo** [ImageMagick/ImageMagick#5883] [Help] [answered]
Hello experts,

I have a problem with image conversions. 

The images are always RGB JPG with AdobeRGB profile included.
Version: ImageMagick 7.1.0-54 Q16-HDRI x86_64 20660
Mac OS Catalina 10.15.7

For this I use the following line:
`magick "R012.jpg" -resample 300 -type TrueColor -profile /Users/xxx/Desktop/Profile/sRGB-IEC61966-2.1.icc "R012.png"`
So far this works very satisfactorily with a lot of images.
For example with this one:
```
magick identify /Users/xxx/Desktop/xxx/jpg_input/R012.jpg 
/Users/xxx/Desktop/xxx/jpg_input/R012.jpg JPEG 3472x4242 3472x4242+0+0 8-bit sRGB 4.0616MiB 0.000u 0:00.001
```

This image:
```
magick identify /Users/xxx/Desktop/xxx/jpg_input/R014.jpg 
/Users/xxx/Desktop/xxx/jpg_input/R014.jpg JPEG 3472x4242 3472x4242+0+0 8-bit sRGB 4.0408MiB 0.010u 0:00.003
```
Gets as error message like this:
`magick: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `/Users/xxx/Desktop/xxx/png_output-direct1/R014.png' @ warning/png.c/MagickPNGWarningHandler/1750`

A .png is created, but it is not an sRGB as desired, but grayscale without profile. 
The desired image is an sRGB RGB image.
All input images are always the same.
How can I adjust the command so that no grayscale .PNG but force that a sRGB .PNG is created?

Thank you for any ideas.
many greetings
pixelzooo

**Comment by fmw42** [comment]
Post the image that does not work.  Does the image look gray?  Does it have the same Adobe profile?

You might want to convert profiles right after reading your input.

**Comment by snibgo** [comment]
@pixelzooo: If an image with an embedded sRGB profile is written to PNG, the profile is not normally written. Instead, a "sRGB" chunk is written. If you want the file to contain the profile instead of the chunk, use "`-define png:exclude-chunk=sRGB`". See https://imagemagick.org/script/defines.php

If this doesn't fix your problem, please link to sample image files.

**Comment by pixelzooo** [comment]
Hello @fmw42  and @snibgo! 
Thank you very much for the feedback. 
The input images are always RGB with AdobeRGB ICC profile. Yes the images look gray. At which place in my call `magick "R012.jpg" -resample 300 -type TrueColor -profile /Users/xxx/Desktop/Profile/sRGB-IEC61966-2.1.icc "R012.png"` do I have to insert this command from you `-define png:exclude-chunk=sRGB`? Or is the command not given with every call but set once globally? How exactly do I have to set it? Thank you so much.

**Reply by fmw42** [reply]
Please post an example image.  If the image is gray with 3 equal channels, then some tools will still call it RGB. But Imagemagick will say it is gray.  If you have 3 equal channels and have an associated sRGB or Adobe profile, then that may be why IM says there is an issue. If you want PNG output to be RGB rather than gray, preface with PNG24:, that is specify PNG24:output.png. If the PNG is transparent, then PNG32:

Best if you provide and example image.

**Comment by pixelzooo** [comment, marked as answer]
hi, thanks for the great help and the information the specification `PNG24:output.png` has solved the problem. THANKS!
